aten/src/ATen/native/PixelShuffle.cpp

```diff
@@ -51,4 +51,49 @@ Tensor pixel_shuffle(const Tensor& self, int64_t upscale_factor) {
   return input_permuted.reshape(final_shape);
 }
 
+
+Tensor pixel_unshuffle(const Tensor& self, int64_t downscale_factor) {
+  TORCH_CHECK(self.dim() >= 3,
+              "pixel_unshuffle expects input to have at least 3 dimensions, but got input with ",
+              self.dim(), " dimension(s)");
+  // Format: (B1, ..., Bn), C, H, W
+  int64_t c = self.size(-3);
+  int64_t h = self.size(-2);
+  int64_t w = self.size(-1);
+  const auto NUM_NON_BATCH_DIMS = 3;
+  const auto last_batch_dim = self.sizes().end() - NUM_NON_BATCH_DIMS;
```

Comment:
nit: I would definitely expect something called `last_batch_dim` to be an integer type, that it's only valid on self_sizes should probably be part of the name.

Reply:
also, doesn't this point to the first non-batch dim?  I.e. it's only the last_batch_dim because you are calling it in a function that "ends" one past.  So maybe a better name is "self_sizes_batch_end" or something.

Reply:
Yes, you're right- it points to just after the last batch dim. To be more precise, I'll change this to `self_sizes_batch_end` as suggested.